include/seqan3/io/sam_file/format_bam.hpp

```diff
@@ -333,24 +333,43 @@ inline void format_bam::read_alignment_record(stream_type & stream,
         if (!std::ranges::equal(stream_view | views::take_exactly_or_throw(4), std::string_view{"BAM\1"}))
             throw format_error{"File is not in BAM format."};
 
-        int32_t tmp32{};
-        read_field(stream_view, tmp32);
+        int32_t l_text{}; // length of header text including \0 character
+        int32_t n_ref{}; // number of reference sequences
+        int32_t l_name{}; // 1 + length of reference name including \0 character
+        int32_t l_ref{}; // length of reference sequence
```

Comment:
Per specification, these should be `uint32_t`. Maybe there was a reason they are not, and maybe we can find that reason.
Just replacing does not work.

Reply:
Yeah, definitely a follow-up!